src/support/json.h

```diff
@@ -305,20 +311,40 @@ struct Value {
       skip();
       setObject();
       while (*curr != '}') {
-        assert(*curr == '"');
-        curr++;
-        char* close = strchr(curr, '"');
-        assert(close);
-        *close = 0; // end this string, and reuse it straight from the input
-        IString key(curr);
-        curr = close + 1;
-        skip();
-        assert(*curr == ':');
-        curr++;
-        skip();
-        Ref value = Ref(new Value());
-        curr = value->parse(curr);
-        (*obj)[key] = value;
+        if (*curr == '"') {
+          curr++;
+          char* close = curr;
+          while (*close && *close != '"') {
+            if (*close == '\\' && *(close + 1)) {
+              close++; // Skip escaped character
+            }
+            close++;
+          }
```

Comment:
This pattern of skipping escaped characters appears more than once. How about adding a helper function `findEndOfQuoted()`?

Reply:
While doing so, the asserts at the end (next line) could be error checks with `Fatal() << "error message"` (so they also work in non-debug builds - asserts are debug-only).

Reply:
I added 2 helpers using macros : (used macros to stay consistent with the style of this file)
- pattern of skipping chars 
- runtime assert (to take in account release builds), also I am guessing that we need to change all `assert(..)` occurrences in `parse` function at least, since all of them are required ?

Now I have a doubt about the "quoted unquoted" keys in json, while testing I used the file referenced [here](https://github.com/WebAssembly/binaryen/pull/7175#discussion_r1894409313). the said file have keys unquoted, this is allowed by most of the tools out there (browser included) but this doesn't seem to be allowed by JSON RFC.

I added 2 tests to cover the PR.

Reply:
I'm not sure about the JSON spec, but it makes sense for us (as just another tool) to support patterns that are commonly supported by tools, even if the spec isn't clear there.